source/server/hot_restart.proto

```diff
@@ -0,0 +1,61 @@
+syntax = "proto3";
+
+package envoy;
+
+message HotRestartMessage {
+  // Child->parent requests
+  message Request {
+    message PassListenSocket {
+      string address = 1;
+    }
+    message ShutdownAdmin {
+    }
+    message Stats {
+    }
+    message DrainListeners {
+    }
+    message Terminate {
+    }
+    oneof request {
+      PassListenSocket pass_listen_socket = 1;
+      ShutdownAdmin shutdown_admin = 2;
+      Stats stats = 3;
+      DrainListeners drain_listeners = 4;
+      Terminate terminate = 5;
+    }
+  }
+
+  // Parent->child replies
+  message Reply {
+    message PassListenSocket {
+      int32 fd = 1;
+    }
+    message ShutdownAdmin {
+      uint64 original_start_time = 1;
```

Comment:
specify units in variable name (e.g. _sec or _ms), or use a proto-buffer type for absolute time.

Reply:
Done.